Merge Hash-like values in sample data (#1436)
appsignal/appsignal-ruby: .changesets/merge-metadata-when-comparing-hash-like-values-.md

---
bump: patch
type: fix
---

> When using the transaction helpers to set metadata like `add_params`, `add_sesion_data`, etc.), we will now merge Hash-like values, value types that are subclasses of the Ruby Hash class.
> Now, if a `ActiveSupport::HashWithIndifferentAccess` or `Sinatra::IndifferentHash` value is given, it will be merged with the existing Hash-like value instead of replacing the original value.

Reviewer: Might wanna update this to reflect the new behaviour.

```suggestion
When a hash-like value (such as `ActiveSupport::HashWithIndifferentAccess` or `Sinatra::IndifferentHash`) is passed to a transaction helper (such as `add_params`, `add_session_data`, ...) convert it to a Ruby `Hash` before setting it as the value or merging it with the existing value. 
```
